config/platform.go

```diff
@@ -47,6 +48,56 @@ func readGPUTags(logger logging.Logger, tags []string) []string {
 	return tags
 }
 
+type piModel struct {
+	version     string
+	longVersion string
+}
+
+// inner logic for pi version parsing.
+func parsePi(raw []byte) *piModel {
+	if match := piModelRegex.FindSubmatch(raw); match != nil {
+		litePlus := string(match[3])
+		cm := string(match[1])
+		model := strings.Replace(string(match[5]), " Plus", "p", 1)
+		ret := &piModel{
+			version: string(match[2]),
+		}
+		if cm != "" {
+			ret.longVersion = "cm"
+		}
+		if ret.version == "" {
+			ret.version = "1"
+		}
+		ret.longVersion += ret.version
+		ret.version = ret.version[:1] // contract 3E to 3 now that it's been copied to longVersion
+		switch litePlus {
+		case "Lite":
+			ret.longVersion += "l"
+		case "Plus":
+			ret.longVersion += "p"
+		}
```

Comment:
Is there a default case we haven't hit in all our pis?

Reply:
you mean should this switch stmt have a default?

Reply:
yep to *

Reply:
added warning in bd228f1b3